src/MonoMod.Core/Platforms/Runtimes/Core60Runtime.cs

```diff
@@ -106,6 +107,12 @@ private unsafe void InstallNativeJitHook(IntPtr jit)
             MemoryMarshal.Write(ptrData, ref ourCompileMethodHookPtr);
 
             System.PatchData(PatchTargetKind.ReadOnly, (IntPtr)compileMethodSlot, ptrData, default);
+
+            // now trigger the compilation of something to ensure that any patches deferred to first invoke are performed
+            var dm = new DynamicMethod("CompileMethodPatchPrimer", typeof(void), null, typeof(Core60Runtime), false);
+            var il = dm.GetILGenerator();
+            il.Emit(OpCodes.Ret);
+            dm.Invoke(null, null);
```

Comment:
I think we can (and probably should) use a single-callsite `NoInlining` method for this, or at least use a DMD here instead of DynamicMethod directly.